views/default/navigation/menu/elements/item/url.php

```diff
@@ -0,0 +1,39 @@
+<?php
+/**
+ * Output an ElggMenuItem as an url
+ *
+ * @uses $vars['item'] the ElggMenuItem to display
+ */
+
+$item = elgg_extract('item', $vars);
+if (!$item instanceof \ElggMenuItem) {
+	return;
+}
+
+unset($vars['item']);
+
+$badge = $item->getData('badge');
```

Comment:
can we change this to logic that happens on $vars? no need to setBadge back i thinks

Reply:
sure....